File: online-store-api/online-store-api/src/main/java/com/code81/onlinestore/service/StaffService.java
package com.code81.onlinestore.service;

import com.code81.onlinestore.dto.common.PageResponse;
import com.code81.onlinestore.dto.staff.StaffCreateRequest;
import com.code81.onlinestore.dto.staff.StaffResponse;
import com.code81.onlinestore.dto.staff.StaffUpdateRequest;
import org.springframework.data.domain.Pageable;

public interface StaffService {

    StaffResponse create(StaffCreateRequest request);

    StaffResponse update(Long id, StaffUpdateRequest request);

    StaffResponse getById(Long id);

    PageResponse<StaffResponse> list(Pageable pageable);
}

File: code81_online_store_angular/src/app/core/services/staff.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { StaffCreateRequest, StaffResponse, StaffUpdateRequest } from '../models/staff.model';
import { PageResponse } from '../models/common.model';

@Injectable({ providedIn: 'root' })
export class StaffService {
  private http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiUrl}/staff`;

  list(page = 0, size = 20): Observable<PageResponse<StaffResponse>> {
    const params = new HttpParams().set('page', page).set('size', size);
    return this.http.get<PageResponse<StaffResponse>>(this.baseUrl, { params });
  }

  getById(id: number): Observable<StaffResponse> {
    return this.http.get<StaffResponse>(`${this.baseUrl}/${id}`);
  }

  create(request: StaffCreateRequest): Observable<StaffResponse> {
    return this.http.post<StaffResponse>(this.baseUrl, request);
  }

  update(id: number, request: StaffUpdateRequest): Observable<StaffResponse> {
    return this.http.put<StaffResponse>(`${this.baseUrl}/${id}`, request);
  }
}
